core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -134,6 +135,8 @@ class ReplicaFetcherThread(name: String,
       if (logger.isTraceEnabled)
         trace("Follower %d set replica high watermark for partition [%s,%d] to %s"
           .format(replica.brokerId, topic, partitionId, followerHighWatermark))
+      if (quota.isThrottled(topicAndPartition))
+        quota.record(partitionData.toByteBufferMessageSet.sizeInBytes)
```

Comment:
Would it be better to just use `messageSet` from line 118 instead of doing `partitionData.toByteBufferMessageSet` again? This way you save an allocation.